### PR TITLE
Make port number optional for logical Load Balancer 

### PR DESCRIPTION
**Pull request by SanikaGawhane**
# Description

Added unit and integration tests.

## Related PR and Issues
Fixes # [https://github.com/cloudfoundry/bosh-vsphere-cpi-release/issues/255]

## Impacted Areas in Application
List general components of the application that this PR will affect:

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Environment Variables for Integration test:
* Hardware Requirements in ESXi:
* Toolchain:
* SDK:

# Checklist:

- [ ] My code follows the standard ruby style guide
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/vsphere_cpi/spec/integration/spec_helper.rb [hunk 1]
```diff
@@ -43,7 +43,7 @@
 
   rspec_config.after(:suite) do
     if ENV.fetch('BOSH_VSPHERE_STEMCELL_ID', '').empty?
-      delete_stemcell(suite_cpi, stemcell_id)
+      #delete_stemcell(suite_cpi, stemcell_id)
```
Comment (EleanorRigby) [inline src/vsphere_cpi/spec/integration/spec_helper.rb]: This needs to be removed

Comment (EleanorRigby) [reply src/vsphere_cpi/spec/integration/spec_helper.rb]: the comment `#` syntax

File: src/vsphere_cpi/spec/integration/nsxt_spec.rb [hunk 2]
```diff
@@ -531,10 +531,10 @@ def create_dynamic_server_pool(pool_name, nsgroup)
     services_svc.create_load_balancer_pool(server_pool)
   end
 
-  def find_pool_members(server_pool, ip_address, port_no)
+  def find_pool_members(server_pool, ip_address, *port_no)
     server_pool = services_svc.read_load_balancer_pool(server_pool.id)
     server_pool.members.select do |member|
-      member.ip_address == ip_address
+      member.ip_address == ip_address || (member.ip_address == ip_address && member.port == port_no)
```
Comment (EleanorRigby) [inline src/vsphere_cpi/spec/integration/nsxt_spec.rb]: The second part of || on right is useless

File: src/vsphere_cpi/spec/integration/nsxt_spec.rb [hunk 3]
```diff
@@ -239,11 +239,9 @@
               'server_pools' => [
                 {
                   'name' => server_pool_name_1,
-                  'port' => port_no
```
Comment (EleanorRigby) [inline src/vsphere_cpi/spec/integration/nsxt_spec.rb]: This tests without port number. We need a test with port number also.

File: src/vsphere_cpi/spec/integration/nsxt_spec.rb [hunk 4]
```diff
@@ -407,7 +405,7 @@
 
           server_poo1_1_members = find_pool_members(server_pool_1, vm_ip, port_no1)
           server_poo1_1_members.concat(find_pool_members(server_pool_1, vm_ip, port_no2))
-          expect(server_poo1_1_members.count).to eq(2)
+          expect(server_poo1_1_members.count).to eq(4)
```
Comment (EleanorRigby) [inline src/vsphere_cpi/spec/integration/nsxt_spec.rb]: Why will this change to 4?

**Comment by julian-hj** [comment]
removed stale label since we probably want most of the changes in this PR

**Comment by markstokan** [comment]
Closing.

The BOSH vSphere CPI code has always handled the absence of a port parameter gracefully.  We have added [tests](https://github.com/cloudfoundry/bosh-vsphere-cpi-release/commit/0ce6fc6abc7a3dc0e9237efc58550d4fc5a917ce) using the manager and policy api to draw attention to this.

An [issue](https://github.com/pivotal-cf/ops-manager/issues/411) is open with the Ops Manager team to make the Port optional when configuring this through the Resource Config UI.

**Pull request closed without merging**